File: Desafio Mobile iOS/Desafio Mobile iOS/Models/Owner.m
//
//  Owner.m
//  Desafio Mobile iOS
//
//  Created by Adriano Rezena on 10/10/17.
//  Copyright © 2017 Adriano Rezena. All rights reserved.
//

#import "Owner.h"

@implementation Owner

-(id)initWithDictionary:(NSDictionary *)dictionary {
    if (self = [super init]) {
        self.login = dictionary[@"login"];
        self.avatar_url = dictionary[@"avatar_url"];
    }
    
    return self;
}

@end

File: DesafioIOS/DesafioIOS/Models/Owner.swift
//
//  Owner.swift
//  DesafioIOS
//
//  Created by Felipe Ricieri on 28/10/17.
//  Copyright © 2017 Nexaas. All rights reserved.
//

import Foundation

class Owner : NSObject {
    
    var id : String = ""
    var name : String = ""
    var username : String = ""
    var picture : String = ""
    
    init(data: [String:Any]) {
        
        self.id = (data["id"] as? String).unwrapOrElse("")
        self.name = (data["login"] as? String).unwrapOrElse("")
        self.username = (data["login"] as? String).unwrapOrElse("")
        self.picture = (data["avatar_url"] as? String).unwrapOrElse("")
    }
}

